### DISCUSSION TITLE
Use this project to monitor gas water heater on/off

### DISCUSSION THREAD

**Discussion by 42Network** [tronikos/esphome-magnetometer-water-gas-meter#9] [General]
I'm interested in monitoring when my water heater turns on and off. Mine has a basic thermopile-powered Honeywell controller which uses a solenoid to open the gas valve when heat is called for. My iPhone magnetomer tells me there is a distinct step whenever this valve opens or closes. I suspect I could capture this with another ESP32 and magnetomer. It wouldn't be a cyclical pattern like a gas or water meter, though, so it would be a much simpler ESPhome yaml config.

Do you possibly already have something like this?

Thanks!

**Comment by tronikos** [comment]
No I don't

**Comment by jac2424** [comment]
Perhaps you can do some rough calculation to see how much gas flow increases when hot water heater is running and crated a home assistant templates binary sensor to show on when the pattern matches. I was considering doing something similar but have not gotten around to it yet as I am still trying to get the flow data to read accurately consistently.
